src/Symfony/Component/Form/FormFactory.php

```diff
@@ -58,11 +58,30 @@ public function createForProperty($class, $property, $data = null, array $option
     /**
      * {@inheritdoc}
      */
-    public function createBuilder($type = 'form', $data = null, array $options = array())
+    public function createBuilder($type = 'Symfony\Component\Form\Extension\Core\Type\FormType', $data = null, array $options = array())
     {
-        $name = $type instanceof FormTypeInterface || $type instanceof ResolvedFormTypeInterface
-            ? $type->getName()
-            : $type;
+        // BC
+        if ($type instanceof FormTypeInterface || $type instanceof ResolvedFormTypeInterface) {
+            $typeName = $type->getName();
+        } else {
+            $typeName = $type;
+        }
+
+        if (false === ($pos = strrpos($typeName, '\\'))) {
+            // No FQCN - leave unchanged for BC
+            $name = $typeName;
+        } else {
+            // FQCN - strip namespace
+            $name = substr($typeName, $pos + 1);
+
+            // strip "Type" suffix
+            if (strlen($name) > 4 && 'type' === strtolower(substr($name, -4))) {
```

Comment:
It think would be fast to use http://php.net/manual/en/function.substr-compare.php which has case insensitivity optionally built into it and removes the need to create a new string with substr.